doc/api/domain.markdown

```diff
@@ -2,304 +2,265 @@
-  // resources like crazy if this ever happens.
-  // This is no better than process.on('uncaughtException')!
-  console.log('error, but oh well', er.message);
-});
-d.run(() => {
-  require('http').createServer((req, res) => {
-    handleRequest(req, res);
-  }).listen(PORT);
-});
-```
-
-By using the context of a domain, and the resilience of separating our
-program into multiple worker processes, we can react more
-appropriately, and handle errors with much greater safety.
-
-```javascript
-// Much better!
-
-const cluster = require('cluster');
-const PORT = +process.env.PORT || 1337;
-
-if (cluster.isMaster) {
-  // In real life, you'd probably use more than just 2 workers,
-  // and perhaps not put the master and worker in the same file.
-  //
-  // You can also of course get a bit fancier about logging, and
-  // implement whatever custom logic you need to prevent DoS
-  // attacks and other bad behavior.
-  //
-  // See the options in the cluster documentation.
-  //
-  // The important thing is that the master does very little,
-  // increasing our resilience to unexpected errors.
-
-  cluster.fork();
-  cluster.fork();
-
-  cluster.on('disconnect', (worker) => {
-    console.error('disconnect!');
-    cluster.fork();
-  });
-
-} else {
-  // the worker
-  //
-  // This is where we put our bugs!
-
-  const domain = require('domain');
-
-  // See the cluster documentation for more details about using
-  // worker processes to serve requests.  How it works, caveats, etc.
-
-  const server = require('http').createServer((req, res) => {
-    var d = domain.create();
-    d.on('error', (er) => {
-      console.error('error', er.stack);
-
-      // Note: we're in dangerous territory!
-      // By definition, something unexpected occurred,
-      // which we probably didn't want.
-      // Anything can happen now!  Be very careful!
-
-      try {
-        // make sure we close down within 30 seconds
-        var killtimer = setTimeout(() => {
-          process.exit(1);
-        }, 30000);
-        // But don't keep the process open just for that!
-        killtimer.unref();
-
-        // stop taking new requests.
-        server.close();
-
-        // Let the master know we're dead.  This will trigger a
-        // 'disconnect' in the cluster master, and then it will fork
-        // a new worker.
-        cluster.worker.disconnect();
-
-        // try to send an error to the request that triggered the problem
-        res.statusCode = 500;
-        res.setHeader('content-type', 'text/plain');
-        res.end('Oops, there was a problem!\n');
-      } catch (er2) {
-        // oh well, not much we can do at this point.
-        console.error('Error sending 500!', er2.stack);
+The better approach is to handle thrown errors gracefully by sending an error
+response to the request that triggered the error, while letting other requests
+finish in their normal time.
+
+The `domain` module can work well with the `cluster` module. For instance,
```

Comment:
I would add something lie "For this specific use case", or something that makes clear that when using the domain module, one doesn't have to or shouldn't automatically think about the cluster module.